src/sentry/issues/auto_source_code_config/task.py

```diff
@@ -174,57 +175,69 @@ def create_repos_and_code_mappings(
 
     organization_id = organization_integration.organization_id
     for code_mapping in code_mappings:
-        repository = (
-            Repository.objects.filter(name=code_mapping.repo.name, organization_id=organization_id)
-            .order_by("-date_added")
-            .first()
+        process_code_mapping(
+            organization_id,
+            organization_integration,
+            project,
+            code_mapping,
+            platform,
+            dry_run,
         )
 
-        if not repository:
-            if not dry_run:
-                repository = Repository.objects.create(
-                    name=code_mapping.repo.name,
-                    organization_id=organization_id,
-                    integration_id=organization_integration.integration_id,
-                )
-            metrics.incr(
-                key=f"{METRIC_PREFIX}.repository.created",
-                tags={"platform": platform, "dry_run": dry_run},
-                sample_rate=1.0,
-            )
 
-        extra = {
-            "project_id": project.id,
-            "stack_root": code_mapping.stacktrace_root,
-            "repository_name": code_mapping.repo.name,
-        }
-        # The project and stack_root are unique together
-        existing_code_mappings = RepositoryProjectPathConfig.objects.filter(
-            project=project, stack_root=code_mapping.stacktrace_root
-        )
-        if existing_code_mappings.exists():
-            logger.warning("Investigate.", extra=extra)
-            continue
+def process_code_mapping(
+    organization_id: int,
+    organization_integration: RpcOrganizationIntegration,
+    project: Project,
+    code_mapping: CodeMapping,
+    platform: str,
+    dry_run: bool,
+) -> None:
+    repository = (
+        Repository.objects.filter(name=code_mapping.repo.name, organization_id=organization_id)
+        .order_by("-date_added")
+        .first()
+    )
 
+    if not repository:
         if not dry_run:
-            if repository is None:  # This is mostly to appease the type checker
```

Comment:
I have changed the code in such a way that we don't need to make this specific check:
<img width="781" alt="image" src="https://github.com/user-attachments/assets/8de16927-f516-48dc-b460-e0aea2dab5b1" />